org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/migration/DoStructureMigrationInventory.java

```diff
@@ -199,10 +213,17 @@ protected void init() {
       versions.sort(m_comparator);
       m_typeNameVersions.put(entry.getKey(), versions);
     }
+
+    // Collect value migration handlers
+    getAllValueMigrationHandlers().stream()
+        .sorted(Comparator
+            .comparing(handler -> BEANS.get(((IDoValueMigrationHandler<?>) handler).typeVersion()).getVersion(), m_comparator) // primary sort order: type version
```

Comment:
BEANS.get was called on the Type Version. Changed implementation of IDoValueMigrationHandler and AbstractDoValueMigrationHandler to match existing pattern in IDoStructureMigrationHandler and AbstractDoStructureMigrationHandler: typeVersion() method returns a NamespaceVersion instead of Class<? extends ITypeVersion>.